### PR TITLE
[ Core ]Fix bug for set gcs port environment variable

### PR DESCRIPTION
**Pull request by XiaodongLv**
## Why are these changes needed?
When RAY_GCS_SERVER_PORT is set, ray.init() is executed locally will failed.
TypeError: '>' not supported between instances of 'str' and 'int'.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/BUILD [hunk 1]
```diff
@@ -304,6 +304,7 @@ py_test_module_list(
     "test_usage_stats.py",
     "test_placement_group_3.py",
     "test_placement_group_5.py",
+    "test_env.py",
```
Comment (jovany-wang) [inline python/ray/tests/BUILD:307]:
```suggestion
    "test_set_gcs_server_port.py",
```

File: python/ray/_private/node.py [hunk 2]
```diff
@@ -270,7 +270,16 @@ def __init__(
 
         # Pick a GCS server port.
         if head:
-            gcs_server_port = os.getenv(ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)
+            gcs_server_port_environment = os.getenv(
```
Comment (jjyao) [inline python/ray/_private/node.py:273]: ray_constants.py has a helper function called `env_integer()`, lets use that.

Comment (XiaodongLv) [reply python/ray/_private/node.py:273]: Thankyou, I will change it.

Comment (XiaodongLv) [reply python/ray/_private/node.py]: @jjyao I have completed it

File: python/ray/_private/node.py [hunk 3]
```diff
@@ -270,7 +270,11 @@ def __init__(
 
         # Pick a GCS server port.
         if head:
-            gcs_server_port = os.getenv(ray_constants.GCS_PORT_ENVIRONMENT_VARIABLE)
+            from ray.autoscaler._private.constants import env_integer
```
Comment (rkooo567) [inline python/ray/_private/node.py:273]: can you use the method under ray._private.ray_constants, not the autoscaler package? (not sure why it has the same method there)

File: python/ray/tests/test_set_gcs_server_port.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+def test_as_gcs_server_port_is_set():
+    port_int = random.randint(6000, 10000)
+    port_str = str(port_int)
+    os.environ["RAY_GCS_SERVER_PORT"] = port_str
+    ray.init()
+    assert detect_listened_port(port_int)
+    ray.shutdown()
+
+
+# If RAY_GCS_SERVER_PORT is set as string, it will be ignored so that
```
Comment (rkooo567) [inline python/ray/tests/test_set_gcs_server_port.py:49]: The comment and test doesn't seem to match?

**Comment by jovany-wang** [comment]
Duplicate to #34482 . closing

**Pull request closed without merging**